File: SNM.FrontEnd/src/app/core/models/InstagramImage.ts
import { MediaType } from "../enumerations/MedyaType";
import { IInstagramImage } from "../types/instagramImage.types";

export class InstagramImage implements Required<IInstagramImage> {
    id: string | null;
    media_type :MediaType 
    instagramPostlId:string
    media_url:string
  
    constructor(instagramImage: IInstagramImage) {
      this.id = instagramImage.id || null;
      this.media_type=instagramImage.media_type;
      this.media_url=instagramImage.media_url;
      this.instagramPostlId=instagramImage.instagramPostlId;
  
     
    }
  }

File: Instagram/Core/Instagram.Domain/Entities/InstagramImage.cs
using SNM.Instagram.Domain.Common;
using SNM.Instagram.Domain.Enumeration;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SNM.Instagram.Domain.Entities
{
    public class InstagramImage:EntityBase<Guid>
    {
        public MediaTypeEnum Media_type { get; set; } //Video or image   
        public string Media_url { get; set; } // video or image
        public virtual InstagramPost? InstagramPost { get; set; }
        public Guid InstagramPostlId { get; set; }

    }
}
